apps/prenumerera/src/Page/Payment.purs

```diff
@@ -44,11 +50,25 @@ component :: Component Props
 component = do
   poller <- Poller.new
   today <- Now.nowDate
-  React.component "Payment" $ \ { user, package, description, offer, method, next } -> React.do
+  globalWindow <- Web.HTML.window
+  React.component "Payment" $ \ { user, package, description, offer, method, next, window } -> React.do
     orderState /\ setOrderState <- useState' $ Right OrderUnknownState
     netsUrl /\ setNetsUrl <- useState' Nothing
     error /\ setError <- useState' $ Right unit
     paperInvoiceConfirmed /\ setPaperInvoiceConfirmed <- useState' false
+    let needTerminal = method == CreditCard &&
+                       (error *> orderState) == Right OrderCreated &&
+                       isJust netsUrl
+        paymentUrl = (maybe "" _.paymentTerminalUrl netsUrl)
+    useEffect needTerminal do
+      if needTerminal then case window of
```

Comment:
I think `when` would read better.

This may be code I wrote myself but I'll comment regardless.

Reply:
I think the original was yours, I just made it worse.